src/mixer/softmixer.rs

```diff
@@ -0,0 +1,56 @@
+use super::Mixer;
+use super::StreamEditor;
+use std::borrow::Cow;
+use std::sync::{Arc, RwLock};
+
+pub struct SoftMixer {
+  volume: Arc<RwLock<u16>>
+}
+
+impl SoftMixer {
+    pub fn new() -> SoftMixer {
+        SoftMixer {
+            volume: Arc::new(RwLock::new(0xFFFF))
+        }
+    }
+}
+
+impl Mixer for SoftMixer {
+    fn init(&self) {
+    }
+    fn start(&self) {
+    }
+    fn stop(&self) {
+    }
+    fn volume(&self) -> u16 {
+        *self.volume.read().unwrap()
+    }
+    fn set_volume(&self, volume: u16) {
+        *self.volume.write().unwrap() = volume;
+    }
+    fn get_stream_editor(&self) -> Option<Box<StreamEditor + Send>> {
+        let vol = self.volume.clone();
+        Some(Box::new(SoftVolumeApplier { get_volume: Box::new(move || *vol.read().unwrap() ) }))
+    }
+}
+
+struct SoftVolumeApplier {
+  get_volume: Box<Fn() -> u16 + Send>
```

Comment:
The privacy boundary is usually the module. As long as it isn't possible to modify the volume unexpectedly from outside the module, that's fine.
I find the closure a lot more confusing than storing the `Arc<RwLock<u16>>`